Add kubernetes deployment yamls

elrickvm opened the pull request:
Kubernetes deployment yamls are needed to stand up the cluster from scratch. Here we will create both the Deployments and Services necessary for internal pod to pod communication and external communication into the cluster. On Google Container Engine we created a static IP address that is used for the External Load Balancer (ELB) Service for entry into the the cluster. That static IP is necessary to give a constant IP address so during Service creation Kubernetes can link the External Load Balancer to that IP.

elrickvm committed:
Create initial Kubernetes deployment and service yaml files

These are Kubernetes deployment and service files that will create the essential modules that will satisfy the basic Okapi / Folio setup on Google Container Engine.

elrickvm committed:
Add static ip to service

A static ip is needed for the okapi gateway so that we can have a reliable address to connect to. We can also point a URL and all the modules at this static ip. During the Kubernetes Service creation it will use this static ip address that is reserved in Google Container Engine.

Commit:
Add RMAPI secrets to mod-kb-ebsco

wwilsman commented on kubernetes/deployments/okapi-deployment.yaml:
```diff
@@ -0,0 +1,98 @@
+          value: "5432"
+        - name: DB_DATABASE
+          value: folio
+        - name: DB_USER
```
This should be `DB_USERNAME` like the others. If I remember correctly, the current custom docker file uses this var (`DB_USER`). The `okapi-initdb-and-start` script will look for `DB_USERNAME` instead of `DB_USER`.

elrickvm replied:
Great catch!

elrickvm committed:
Fix env variable naming

elrickvm committed:
Create an ingress object and ingress controller using nginx

To use SSL/TLS and get an HTTPS connection we needed to create a L7 LoadBalancer using nginx. Initially creating a LoadBalancer on the Okapi service only created an L4 LoadBalancer which was not able to terminate SSL/TLS connections. An ingress object is used in conjuction with an ingress controller to watch for incoming request and send those request to the Okapi backend service while adding the appropriate TLS certs needed.

Commit:
Add mod-login-saml

elrickvm committed:
Change service to NodePort

Initially we had Google Container Engine generate an external loadBalancer (ELB) using a Kubernetes Service that gave us an external ip address that we used to access the okapi gatewway running in the cluster behind this service. However as the need for SSL/TLS came to the forefront we could not rely on the ELB that Google Container Engine generated with the Service because it was an L4 loadBalancer and L4 cannot terminate SSL/TLS so we need a L7. Changing the port allows this to be configured to work with the nginx ingress that is setup in the cluster.
